src/spect_experimental/utilities/ctac_to_mu_values.cxx

```diff
@@ -0,0 +1,224 @@
+//
+//
+/*
+  Copyright (C) 2019, Institute of Nuclear Medicine, University College London
+  This file is part of STIR.
+
+  This file is free software; you can redistribute it and/or modify
+  it under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  This file is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU Lesser General Public License for more details.
+
+  See STIR/LICENSE.txt for details
+*/
+/*!
+  \file
+  \ingroup utilities
+  \brief Produces an image of mu-values from a CT/CTAC.
+  \author Benjamin A. Thomas
+
+  \par Usage:
+  \code
+  ctac_to_mu_values -o output_filename -i input_directory -j slope_filename
+  \endcode
+*/
+
+#include <algorithm>
+#include <cctype>
+
+#include "stir/IO/ITKImageInputFileFormat.h"
+#include "stir/info.h"
+#include "stir/Succeeded.h"
+#include "stir/IO/OutputFileFormat.h"
+#include "stir/IO/read_from_file.h"
+#include "stir/DiscretisedDensity.h"
+#include "stir/modelling/ParametricDiscretisedDensity.h"
+#include "stir/DynamicDiscretisedDensity.h"
+#include "stir/VoxelsOnCartesianGrid.h"
+#include "stir/stream.h"
+#include "stir/getopt.h"
+
+
+#include <nlohmann/json.hpp>
+
+USING_NAMESPACE_STIR
+
+typedef DiscretisedDensity<3,float> FloatImageType;
+
+Succeeded apply_bilinear_scaling_to_HU(
+    const std::unique_ptr<FloatImageType> &input_image_sptr,
+    const nlohmann::json &transform,
+    std::shared_ptr<FloatImageType> output_image_sptr){
+
+  FloatImageType::full_iterator out_iter = output_image_sptr->begin_all();
+  FloatImageType::const_full_iterator in_iter = input_image_sptr->begin_all_const();
+
+  const float a1 = transform["a1"];
+  const float b1 = transform["b1"];
+
+  const float a2 = transform["a2"];
+  const float b2 = transform["b2"];
+
+  //std::cout << transform.dump(4);
+
+  while( in_iter != input_image_sptr->end_all_const())
+  {
+    if (*in_iter<0.f) {
+      float mu = a1 + b1 *(*in_iter);
+      *out_iter = (mu < 0.0f) ? 0.0f : mu;
+    } else {
+      *out_iter = a2 + b2 * (*in_iter);
+    }
+
+    ++in_iter; ++out_iter;
+  }
+
+  return Succeeded::yes;
+}
+
+
+int main(int argc, char * argv[])
+{
+  USING_NAMESPACE_STIR;
+  const char * output_filename = 0;
+  const char * input_filename = 0;
+  const char * slope_filename = 0;
+  const char * manufacturer_name = 0;
+  const char * keV_str = 0;
```

Comment:
I guess we currently need to pass kvp as well (option `-v` for "voltage"?)

Reply:
Done 905a972